lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -113,6 +156,17 @@ EscapeAnalysis::findCachedPointerKind(SILType Ty, const SILFunction &F) const {
   return pointerKind;
 }
 
+EscapeAnalysis::PointerKind
+EscapeAnalysis::findCachedObjectKind(SILType Ty, const SILFunction &F) const {
+  auto iter = objectKindCache.find(Ty);
```

Comment:
Caching the pointer kind sounds dangerous. The pointer kind is computed per function and can be different per function (because the resilience expansion can be different). But it's cached globally for all functions.

Maybe always use ResilienceExpansion::Minimal?

BTW, it's the same situation with the already existing `pointerKindCache`.

Reply:
Thanks, I changed that to Maximal after the last feedback without even thinking.